src/app/project/settings/settings.component.ts

```diff
@@ -388,43 +388,47 @@ export class SettingsComponent extends SubscriptionWrapper implements OnInit {
       this.projectService.createProject(this.companyId, this.project).pipe(
         switchMap((project) => {
           this.project.id = project.id;
-          this.updateCustomer(); // Set Customer
-
-          // Add Company Account to ProjectUser
-          let companyAcc: ProjectUser = {
-            user: {
-              id: this.authService.currentUserValue.id,
-              username: this.authService.currentUserValue.username,
-              lastname: this.authService.currentUserValue.lastname,
-              firstname: this.authService.currentUserValue.firstname,
-            },
-            roles: [this.listOfRoles.find((v) => v.name === 'Firma')],
-          };
-
-          return this.projectUserService.updateProjectUser(
-            project.id,
-            companyAcc.user.id,
-            companyAcc
-          );
-        })
+
+          return forkJoin([
+            this.updateCompanyUser(project.id),
+            this.updateCustomer(project.id),
+          ]);
+        }),
+        tap(() => this.routeToProjectDashboard(this.companyId))
       )
     );
   }
 
-  updateCustomer() {
+  updateCompanyUser(projectId: string): Observable<ProjectUser> {
+    // Add Company Account to ProjectUser
+    let companyAcc: ProjectUser = {
+      user: {
+        id: this.authService.currentUserValue.id,
+        username: this.authService.currentUserValue.username,
+        lastname: this.authService.currentUserValue.lastname,
+        firstname: this.authService.currentUserValue.firstname,
+      },
+      roles: [this.listOfRoles.find((v) => v.name === 'Firma')],
+    };
+
+    return this.projectUserService.updateProjectUser(
+      projectId,
+      companyAcc.user.id,
+      companyAcc
+    );
```

Comment:
Könnte man auch so schreiben 😈
```suggestion
    return this.projectUserService.updateProjectUser(
      projectId,
      this.authService.currentUserValue.id,
      {
      user: {
        id: this.authService.currentUserValue.id,
        username: this.authService.currentUserValue.username,
        lastname: this.authService.currentUserValue.lastname,
        firstname: this.authService.currentUserValue.firstname,
      },
      roles: [this.listOfRoles.find((v) => v.name === 'Firma')],
    }
    );
```

Für kompletten JS Müll (untested):
```js
return this.projectUserService.updateProjectUser(
  projectId,
  companyAcc.user.id,
  {
	user: {
	  ...((({ token, ...rest }) => rest)(this.authService.currentUserValue)),
	  id: this.authService.currentUserValue.id,
	},
	roles: [this.listOfRoles.find((v) => v.name === 'Firma')],
  }
);
```

Reply:
Schreibe ich vor dem Merge noch um 👍